src/components/dialogs/commons/unique-name-input.tsx

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, {
+    ChangeEvent,
+    FunctionComponent,
+    useCallback,
+    useEffect,
+} from 'react';
+import { useDebounce } from '@gridsuite/commons-ui';
+import { elementExists } from '../../../utils/rest-api';
+import { FormattedMessage } from 'react-intl';
+import { useSelector } from 'react-redux';
+import { ReduxState } from '../../../redux/reducer.type';
+import { InputAdornment } from '@mui/material';
+import CheckIcon from '@mui/icons-material/Check';
+import { useController, useFormContext } from 'react-hook-form';
+import CircularProgress from '@mui/material/CircularProgress';
+import TextField from '@mui/material/TextField';
+
+interface UniqueNameInputProps {
+    name: string;
+    label?: string;
+    elementType: string;
+    autoFocus?: boolean;
+    onManualChangeCallback?: () => void;
+}
+
+/**
+ * Input component that constantly check if the field's value is available or not
+ */
+export const UniqueNameInput: FunctionComponent<UniqueNameInputProps> = (
```

Comment:
can you add the custom TextFieldProps, so we can customize this component 
```
interface UniqueNameInputPropsBase {
    name: string;
    label?: string;
    elementType: string;
    onManualChangeCallback?: () => void;
}

type UniqueNameInputProps = UniqueNameInputPropsBase & TextFieldProps;
/**
 * Input component that constantly check if the field's value is available or not
 */
export const UniqueNameInput: FunctionComponent<UniqueNameInputProps> = (
```

Reply:
Ok I will do it but like it's done for other RHF component, with an extra props called `formProps`